src/main/java/org/truffleruby/core/string/FrozenStrings.java

```diff
@@ -50,6 +54,8 @@ public DynamicObject getFrozenString(Rope rope) {
             string = frozenStrings.get(holder);
             if (string == null) {
                 string = StringOperations.createFrozenString(context, rope);
+                // TODO CS 8-Jan-18 as in RopeCache the map is weak, but they key is a wrapper! So we just hold on to everything forever using this method.
```

Comment:
This bug is so bad that we may as well just remove all the weak collections we have, open a Jira, and come back to it later.

Reply:
`but they key` => `but the key`.

Reply:
Fixed.